cpp/include/cudf/lists/lists_column_view.hpp

```diff
@@ -87,6 +87,20 @@ class lists_column_view : private column_view {
    * @throw cudf::logic error if this is an empty column
    */
   column_view get_sliced_child(rmm::cuda_stream_view stream) const;
+
+  /**
+   * @brief Return first offset (accounting for column offset)
+   *
+   * @return int32_t const* Pointer to the first offset
+   */
+  auto const* offsets_begin() const noexcept { return offsets().begin<int32_t>() + offset(); }
```

Comment:
Something like this would be a little cleaner and more canonical. 
```suggestion
  using offset_type = size_type;
  using offset_iterator = offset_type*;
  offset_iterator offsets_begin() const noexcept { return offsets().begin<offset_type>() + offset(); }
```

Reply:
Shouldn't this be `using offset_type = int32_t`?

Reply:
Strictly speaking, the offset_type is based on the `size_type` as it needs to be large enough to index into a `column`.

Reply:
I'm confused now as this seems at odds with https://github.com/rapidsai/cudf/pull/7376#discussion_r575284185

Reply:
Technically I think `using offset_type = int32_t;` is more correct because the offsets column type is actually `INT32`.

Reply:
I guess it doesn't really matter, but I'd still argue that `size_type` is correct as if `size_type` ever changed then so too would the type used for offsets in list and string columns need to change.